### PR TITLE
Feature/#34 implement budget account entity deletion

### LINKED ISSUE #34: Implement `BudgetAccount` entity deletion
Through Edit Budget Account UI - please add button to delete `BudgetAccount` entity

We should be doing `DELETE` request to the `/api/budgetAccount/[id]` route handler - which means we need to add 
`export async function DELETE(request: Request)` into this route handler

From the API route we should validate following:

1. User session is present
2. Budget Account specified in the request parameters has `userId` set to session's user

### PR DESCRIPTION
**Pull request by Artemh59**
Closes #34

### CODE REVIEW

File: src/components/molecules/BudgetAccount/BudgetAccountForm.tsx [hunk 1]
```diff
@@ -203,22 +232,35 @@ export default function BudgetAccountForm({
...
-          <Button
-            type="submit"
-            variant="contained"
-            disabled={isPending}
-          >
-            {account ? 'Edit' : 'Create'}
-          </Button>
+        <DialogActions sx={{ justifyContent: account ? 'space-between' : 'flex-end' }}>
+          {account && (
+            <Button
```
Comment (mudrila) [inline src/components/molecules/BudgetAccount/BudgetAccountForm.tsx:237]: Let's make this `IconButton` with Trash Icon

```
import DeleteOutlineIcon from '@mui/icons-material/DeleteOutline';
```

File: src/components/molecules/BudgetAccount/BudgetAccountForm.tsx [hunk 2]
```diff
@@ -44,9 +45,37 @@ export default function BudgetAccountForm({
   const handleClose = () => {
     setOpen(false);
   };
+
+  const handleDelete = (event: FormEvent) => {
+    event.preventDefault();
+    const loadingToastId = toast.loading(`Hand tight - we are deleting budget account for ya...`);
```
Comment (mudrila) [inline src/components/molecules/BudgetAccount/BudgetAccountForm.tsx:51]: Let's slightly change message here - `Hand tight - we are deleting ${account.title}...`

File: src/app/api/budgetAccount/[id]/route.ts [hunk 3]
```diff
@@ -62,3 +62,36 @@ export async function PUT(request: Request, { params }: { params: { id: string }
...
+    });
+
+    if (!existingAccount) {
+      return NextResponse.json({
+        errorMessage: "You don't have permission to edit this budget account",
+      });
+    }
+
+    await prisma.budgetAccount.delete({
+      where: { id, userId },
```
Comment (mudrila) [inline src/app/api/budgetAccount/[id]/route.ts:88]: We don't need `userId` here - we already made sure that it is relates to the session's user, just filtering by `id` is sufficient

**Review by mudrila — CHANGES_REQUESTED** [review]
Few minor changes - but then we should be good to go!

**Comment by Artemh59** [comment]
we also have problem with forms, they are almost invisible

**Comment by mudrila** [comment]
> we also have problem with forms, they are almost invisible

Noted - I'll fix this in scope of dark / light theme switching

**Review by mudrila — APPROVED** [review]
🔥

**Pull request merged**